#!/usr/bin/env python
PACKAGE = "moving_object_detector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("downsample_scale", int_t, 0, "Downsample scale of calculate move points in 3D", 5, 1, 100)
gen.add("moving_flow_length", double_t, 0, "3D flow length[meter] used to distinguish dynamic points from static points", 0.30, 0.01, 1.0)
gen.add("flow_length_diff", double_t, 0, "Maximum allowable flow length diff used for clustering", 0.05, 0.01, 1.0)
gen.add("flow_start_diff", double_t, 0, "Maximum allowable flow start points coodinate diff used for clustering", 0.1, 0.01, 1.0)
gen.add("flow_radian_diff", double_t, 0, "Maximum allowable angular[rad] between two flows used for clustering", 0.10, 0.01, 1.0)
gen.add("flow_axis_max", double_t, 0, "Maximum flow length used for display 3D flow points as colored pointcloud", 0.5, 0.01, 1.0)
gen.add("matching_tolerance", double_t, 0, "Error tolerance[pixel] used in matching points between left, right, current, previous frames. Disabled when minus", 1.0, -1.0, 30.0)
gen.add("cluster_element_num", int_t, 0, "Minimum number of cluster element treated as moving object", 20, 1, 100)

exit(gen.generate(PACKAGE, "moving_object_detector", "MovingObjectDetector"))
